src/github.com/getlantern/http-proxy/server.go

```diff
@@ -164,12 +183,15 @@ func (s *Server) doServe(listener net.Listener, chListenOn *chan string) error {
 
 	s.httpServer = http.Server{Handler: proxy,
 		ConnState: func(c net.Conn, state http.ConnState) {
-			if state == http.StateActive {
-				select {
-				case q <- c:
-				default:
-					fmt.Print("Oops! the connection queue is full!\n")
-				}
+			switch state {
+			case http.StateActive:
+				cb.Put(c)
+			case http.StateClosed:
```

Comment:
Hijacked connections will never transition to this state, so those will never be purged.
We should also make sure the state changes to close with connection errors and potential panics in the handler goroutines. While this method is cleaner than the previous circular buffer, it has the potential to leak if we don't take precautions.